### PR TITLE
Cortx 33302 - 4 New tests added and 1 test fixed

### PR DESCRIPTION
**Pull request by kachhwahadivya**
# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by kachhwahadivya** [comment]
2408 tests collected in 11.31s 
[collection.log](https://github.com/Seagate/cortx-test/files/9315865/collection.log)
[TEST-44782_TestCsmLoad_20220812045915.log](https://github.com/Seagate/cortx-test/files/9315866/TEST-44782_TestCsmLoad_20220812045915.log)
[TEST-44795_TestResourceLimit_20220812023508.log](https://github.com/Seagate/cortx-test/files/9315867/TEST-44795_TestResourceLimit_20220812023508.log)
[TEST-44796_TestResourceLimit_20220812044422.log](https://github.com/Seagate/cortx-test/files/9315868/TEST-44796_TestResourceLimit_20220812044422.log)
[TEST-44797_TestResourceLimit_20220812041531.log](https://github.com/Seagate/cortx-test/files/9315869/TEST-44797_TestResourceLimit_20220812041531.log)
[TEST-44798_TestResourceLimit_20220812045050.log](https://github.com/Seagate/cortx-test/files/9315872/TEST-44798_TestResourceLimit_20220812045050.log)

File: libs/csm/extensions/csm_prov_ext.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        self.log = logging.getLogger(__name__)
+        self.deploy_lc_obj = ProvDeployK8sCortxLib()
+        self.update_seconds = 60
+        self.ha_obj = HAK8s()
+        
+    def copy_sol_file_local(self):
+        self.log.info("Copy solution.yaml from %s to local path %s", K8S_SCRIPTS_PATH, LOCAL_SOLUTION_PATH)
+        remote_sol_path = K8S_SCRIPTS_PATH + "solution.yaml"
+        solution_path = self.master.copy_file_to_local(remote_path=remote_sol_path,local_path=LOCAL_SOLUTION_PATH)
+        return solution_path
```
Comment (ketanarlulkar) [inline libs/csm/extensions/csm_prov_ext.py:40]: This function already exist. Check copy_sol_file in libs.prov.prov_k8s_cortx_deploy.ProvDeployK8sCortxLib

Comment (kachhwahadivya) [reply libs/csm/extensions/csm_prov_ext.py:40]: this func copies local file to remote and not vice versa. Correct me if m wrong

File: libs/csm/extensions/csm_prov_ext.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import time
+import logging
+from commons.utils import config_utils
+from commons.constants import  K8S_SCRIPTS_PATH, K8S_PRE_DISK
+from commons.constants import LOCAL_SOLUTION_PATH
+from libs.prov.prov_k8s_cortx_deploy import ProvDeployK8sCortxLib
+from libs.ha.ha_common_libs_k8s import HAK8s
+
+class ProvExt():
+    """RestIamUser contains all the Rest API calls for iam user operations"""
```
Comment (RajeshPratapSInghChouhan) [inline libs/csm/extensions/csm_prov_ext.py:27]: please change "RestIamUser"

File: tests/csm/rest/test_csm_load.py [hunk 3]
```diff
@@ -870,47 +870,20 @@ def test_44782(self):
...
-        self.log.info("Run last batch of create csm users")
-        result = self.jmx_obj.run_verify_jmx(
-            jmx_file,
-            threads=req_last,
-            rampup=test_cfg["rampup"],
-            loop=1)
-        assert result, "Errors reported in the Jmeter execution"
+        resp = self.csm_obj.list_csm_users(HTTPStatus.OK, return_actual_response=True)
+        existing_user = len(resp.json()['users'])
+        self.csm_obj.create_multi_csm_user(100, existing_user)
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_csm_load.py:875]: can we take '100' from test config or some const ?

**Review by RajeshPratapSInghChouhan — APPROVED** [review]

**Review by ketanarlulkar — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**